### ISSUE TITLE
Custom script template not working in 4.1.1

### ISSUE THREAD

**Issue by KieranCarden** [godotengine/godot#81307]
### Godot version

4.1.1.stable

### System information

Windows 10 - Godot 4.1.1.stable

### Issue description

Detecting a custom script template seems to have changed in 4.1 and I can no longer get Godot to recognize templates from the `%APPDATA%\Godot\script_templates\` location. This does work in 4.0.3.

I am creating the required directories, as directed from the documentation:
`template_path/node_type/file.extension`
![template_path](https://github.com/godotengine/godot/assets/4044137/6a41418b-674d-44c0-9a08-53aed756a38a)

Working in 4.0.3 - Note: this shows two as I created two directories to test this out. A `Node` directory and `State` directory
![create_script_4_0](https://github.com/godotengine/godot/assets/4044137/65bdbb5d-32e5-4861-811b-06b5755b3f5b)

Dialog in 4.1.1
![create_script_4_1](https://github.com/godotengine/godot/assets/4044137/6563c0db-41a9-40f3-844b-473bfbbbfa25)

### Steps to reproduce

These steps are for the versions I tried, though it may be more accurate to say 4.0.x and 4.1.x
Add a custom script template in the given path for your system following documentation requirements.

For a working example:
Create a project in 4.0.3
Add a script, select the custom template that appears.

For 4.1:
Create a project in 4.1.1
Add a script, notice the custom template does not appear.

### Minimal reproduction project

[4_0_template.zip](https://github.com/godotengine/godot/files/12515648/4_0_template.zip)
[4_1_template.zip](https://github.com/godotengine/godot/files/12515649/4_1_template.zip)

Basically empty projects, the only difference being one was created in 4.0.3 and the other in 4.1.1

Template path for editor templates is global, so nothing special needed for that.

**Comment by jsjtxietian** [comment]
I did some test with Godot v4.2.dev (1d4201e82) - Windows 10.0.19045 - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 3060 (NVIDIA; 31.0.15.3619) - 11th Gen Intel(R) Core(TM) i7-11700K @ 3.60GHz (16 Threads)

It seems to work fine with a my custom node template.

**Comment by YuriSizov** [comment]
> Template path for editor templates is global, so nothing special needed for that.

Maybe you're running 4.1 in the self-contained mode? In that case all editor files will be stored next to the Godot executable.

There could also be an issue with the template file itself, though I've checked with completely empty files and it works. Could you provide an example of your template that fails to load?

![image](https://github.com/godotengine/godot/assets/11782833/c2649a94-bc16-4d92-a788-afa5e2dc1224)

**Comment by KieranCarden** [comment]
> Maybe you're running 4.1 in the self-contained mode? In that case all editor files will be stored next to the Godot executable.

So this might have been the issue, but not exactly the same way described here? Either way it gave me idea to check of what could be wrong and indeed found the issue.

I use one of the Godot managers to organize all my projects and executable versions. However, instead of using the built-in download/install function that the manager(s) offer, I always manually grab the version I want to save in a specific location for my own organization purposes. You can "add custom version" in the manager and point to wherever your saved version is. With 4.1.1 though I for some reason just let the manager grab it for me (feeling lazy that day I guess), and with however the manager loads that version it just doesn't see the script templates directory? I have no idea. Manually adding the version like I normally do resolved my issue though.

Either way, user error. I'm closing with this comment.
